rasa_core/agent.py

```diff
@@ -611,11 +621,15 @@ def visualize(self,
         from rasa_core.training.dsl import StoryFileReader
         """Visualize the loaded training data from the resource."""
 
+        # if the user doesn't provide a max history, we will use the
+        # largest value from any policy
+        max_history = max_history or self._max_history()
```

Comment:
Yes, because no policy will have an exact recollection of what happened before max max steps. If we use the min, there would be policies with a longer recollection (e.g. memoization policy) that could differentiate between two conversations, so we should differentiate between them in the visualization as well.